tests/test_context.py

```diff
@@ -0,0 +1,57 @@
+# Source pytest module
+# Copyright (C) 2020  Ngo Ngoc Duc Huy
+#
+# This file is part of palace.
+#
+# palace is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# palace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with palace.  If not, see <https://www.gnu.org/licenses/>.
+
+"""This pytest module tries to test the correctness of the class Context."""
+
+from palace import Context, Device
+
+
+def test_with_context():
+    """Test if `with` can be used to start a context
+    and is destroyed properly.
+    """
+    dev = Device()
+    context = Context(dev)
+    current = Context.current_context()
+    assert current is None
+    with context:
+        current = Context.current_context()
+        assert current is context
+    current = Context.current_context()
+    assert current is None
+
+
+def test_nested_context():
+    """Test if the context manager returns to the
+    previous context."""
+    dev_1 = Device()
+    dev_2 = Device()
+    ctx_1 = Context(dev_1)
+    ctx_2 = Context(dev_2)
+    current = Context.current_context()
+    assert current is None
```

Comment:
To be concerned later when you finished the API, but for now I still think `assert Context.current_context() is None` is enough.